pkg/ipmi/ocp/bootorder.go

```diff
@@ -93,39 +87,45 @@ func remapSortBootOrder(BootOrder *BootOrder) {
 func updateVPDBootOrder(i *ipmi.IPMI, BootOrder *BootOrder) error {
 	var err error
 	var key string
-	var bootType byte
 	var idx int
-	for _, v := range BootOrder.bootSeq {
+	for _, bootType := range BootOrder.bootSeq {
 		key = fmt.Sprintf("Boot%04d", idx)
-		bootType = v & 0x7
 		if bootType == NETWORK_BOOT {
 			log.Printf("VPD set %s to %s", key, NETBOOTER_CONFIG)
 			BootEntries = append(BootEntries, systembooter.BootEntry{Name: key, Config: []byte(NETBOOTER_CONFIG)})
 			if err = Set(key, []byte(NETBOOTER_CONFIG)); err != nil {
 				return err
 			}
 			idx++
+		} else if bootType == NETWORK_BOOT_IPV6 {
+			log.Printf("VPD set %s to %s", key, NETBOOTER_IPV6_CONFIG)
+			BootEntries = append(BootEntries, systembooter.BootEntry{Name: key, Config: []byte(NETBOOTER_IPV6_CONFIG)})
+			if err = Set(key, []byte(NETBOOTER_IPV6_CONFIG)); err != nil {
+				return err
+			}
+			idx++
 		} else if bootType == LOCAL_BOOT {
 			log.Printf("VPD set %s to %s", key, LOCALBOOTER_CONFIG)
 			BootEntries = append(BootEntries, systembooter.BootEntry{Name: key, Config: []byte(LOCALBOOTER_CONFIG)})
 			if err = Set(key, []byte(LOCALBOOTER_CONFIG)); err != nil {
 				return err
 			}
 			idx++
-		} else if bootType == (INVALID_BOOT & 0x7) {
+		} else if bootType == (INVALID_BOOT) {
 			// No need to write VPD
 		} else {
 			log.Printf("Ignoring unrecognized boot type: %x", bootType)
 		}
 	}
 
-	if BmcUpdatedBootorder {
-		// look for a Booter that supports the given configuration
-		for idx, entry := range BootEntries {
-			entry.Booter = systembooter.GetBooterFor(entry)
-			BootEntries[idx] = entry
-		}
+	// Update the BootEntries with booters to match the new VPD
+	for idx, entry := range BootEntries {
+		entry.Booter = systembooter.GetBooterFor(entry)
+		BootEntries[idx] = entry
 	}
+
+	BmcUpdatedBootorder = true
```

Comment:
I think BmcUpdatedBootorder should be renamed to VpdUpdatedBootorder since it could be updated both by BMC and BIOS. On line 39 can be changed to:
// VpdUpdatedBootorder is true when VPD boot order has been updated
VpdUpdatedBootorder bool = false

Reply:
I agree. It has been 6 months, but I found a hole in this logic when either the BIOS or IPMI updated the boot option and the VPD needed to be updated. As you see here the VPD is always reordered when entering updateVPDBootOrder, so the flag moved here.